### DISCUSSION TITLE
Wix toolset version 4 with Managed BootstrapperApplication (created in .net framwework 4.8)

### DISCUSSION THREAD

**Discussion by harshmodhiya** [wixtoolset/issues#8649] [Questions] [answered]
We had created a Managed Bootstrapper Application with the below version: 

- BA - .NET Framework 4.8.1
- WiX toolset - 3.14

Now, we are in the process of upgrading our WiX project (both Bundle and Setup project) from version 3.14 to version 5 using FireGiant's HeatWave extension in Visual Studio 2022.
 
After upgrading, we are encountering the following errors with the BA project:

`Error 0x800700c1: CreateProcessW failed with return code: 193
Error 0x800700c1: Failed to launch bootstrapper application process: "C:\windows\TEMP\{045AB4E5-D686-48DD-BBD2-3739E109735E}\.ba\UI.dll" -burn.ba.apiver 569705357157400576 -burn.ba.pipe BurnPipe.{18A3EBBA-CAD2-42ED-BA5B-BE90F3F31A7E} {BBFD3A24-083F-4161-9A12-FDE5DB39284A}
Error 0x800700c1: Failed to create bootstrapper application process: C:\windows\TEMP\{045AB4E5-D686-48DD-BBD2-3739E109735E}\.ba\UI.dll
Error 0x800700c1: Failed to start bootstrapper application.`

After encountering an error, we found a document indicating that Managed BA and Custom Action projects should be migrated from .NET 4.8.1 to .NET Core 6+ when transitioning from WiX toolset 3.14 to 5. 
So, we have upgraded both projects to .NET Core 8, and now the executable is working fine without the previous error.
 
**Need help with the below,**

- Here, our major concern is with .NET Core that we need to include .NET desktop runtime within our installer (maybe as a prerequisite for a setup to run), which will increase the size of our executable. Any workaround for it?
- Can WiX toolset 4 or 5 be used with the BA project which is in .NET Framework 4.8.1?
- Do we have a document that specifies the minimum .NET framework supported by all versions of the WiX toolset?

**Comment by robmen** [comment]
> we found a document indicating that Managed BA and Custom Action projects should be migrated from .NET 4.8.1 to .NET Core 6+ when transitioning from WiX toolset 3.14 to 5.

Is that document one of ours? If so, can you share the link to it so we can fix it? That document is wrong.

**Reply by harshmodhiya** [reply]
I have followed [Upgrading v3 Managed Bootstrapper Applications](https://wixtoolset.org/docs/fourthree/faqs/#upgrading-v3-managed-bootstrapper-applications:~:text=v4%20added%20the,when%20v4%20shipped) link.
Also, could you please help me find the answers to my questions?

**Reply by Naveensingla2001** [reply]
> > we found a document indicating that Managed BA and Custom Action projects should be migrated from .NET 4.8.1 to .NET Core 6+ when transitioning from WiX toolset 3.14 to 5.
> 
> Is that document one of ours? If so, can you share the link to it so we can fix it? That document is wrong.

So do we need to migrate to .net core or using .net framework 4.8.1 is still fine?

**Reply by tatva-savan** [reply, marked as answer]
I believe .net framework 4.8.1 is still fine.

**Comment by rsmart8452** [comment]
Including .NET desktop runtime? I haven't found a practical workaround for this. I'm personally not a fan of using .NET Core for my BAs because of this.

Can WiX toolset 4 or 5 be used with the BA project which is in .NET Framework 4.8.1? Presumably you're asking if your WiX 3 BA can be used with 4 or 5? Not easily. The gotcha here is that some of Burn's BA events were renamed and some of the event args changed as well. Go through each event handler in your old BA and work it for the new events... meh, yeah, it can be done. My tutorial briefly mentions the challenges around upgrading, but honestly, it's not likely to be much help to you. https://github.com/rsmart8452/Wix4BurnTutorial/wiki

Do we have a document that specifies the minimum .NET framework supported by all versions of the WiX toolset? The WiX source is the best documentation we have. As you can see in their full framework unit test, the devs are currently targeting 4.7.2. https://github.com/wixtoolset/wix/blob/ce73352b1fa1d4f9cded10a0ee410f2e786bd326/src/ext/Bal/test/examples/FullFramework4MBA/App.config
